paper_search_mcp/academic_platforms/semantic.py

```
@@ -439,6 +439,213 @@ def get_paper_details(self, paper_id: str) -> Optional[Paper]:
+            data = response.json()
+            results = data.get('data', [])
+
+            for item in results:
+                # Related papers come in a "relatedPaper" field
+                related_paper = item.get('relatedPaper')
+                if related_paper:
+                    paper = self._parse_paper(related_paper)
+                    if paper:
+                        papers.append(paper)
+
+        except Exception as e:
```

Ruff (0.14.13)

Do not catch blind exception: `Exception`

(BLE001)